web/src/main/java/se/inera/intyg/webcert/web/service/utkast/UtkastServiceImpl.java

```diff
@@ -209,18 +204,34 @@ public Utkast createNewDraft(CreateNewDraftRequest request) {
     }
 
     /**
-     * Update a utkast with data from an existing certificate.
-     *
+     * @see UtkastService#updateDraftFromCandidate(String, String, String, String)
      * @return {@link SaveDraftResponse}
      */
     @Override
     public SaveDraftResponse updateDraftFromCandidate(String fromIntygId, String fromIntygType, String toUtkastId, String toUtkastType) {
+        Utkast toUtkast = getIntygAsDraft(toUtkastId, toUtkastType);
+        verifyUtkastExists(toUtkast, toUtkastId, toUtkastType,
+            "The draft with id '" + toUtkastId + "' cannot be updated since it could not be found");
 
-        // Get the draft (copy to)
-        Utkast to = getIntygAsDraft(toUtkastId, toUtkastType);
+        return updateDraftFromCandidate(fromIntygId, fromIntygType, toUtkast);
+    }
 
-        verifyUtkastExists(to, toUtkastId, toUtkastType, "The draft to copy candidate information to doesn't exist");
-        verifyAccessToCopyFromCandidate(to);
+    /**
+     * @see UtkastService#updateDraftFromCandidate(String, String, Utkast)
+     * @return {@link SaveDraftResponse}
+     */
+    @Override
+    public SaveDraftResponse updateDraftFromCandidate(String fromIntygId, String fromIntygType, Utkast toUtkast) {
+        Utkast to = toUtkast;
+        String toIntygsId = to.getIntygsId();
+        String toIntygsType = to.getIntygsTyp();
+
+        // Draft must be incomplete and only just created (no saving or updates).
+        if (to.getStatus() != UtkastStatus.DRAFT_INCOMPLETE && to.getVersion() != 0) {
```

Comment:
Borde det inte vara !UtkastStatus.DRAFT_INCOMPLETE.equals(to.getStatus())

Reply:
Visst borde det vara så. Fixat.